crates/resource-resolver/src/lib.rs

```diff
@@ -0,0 +1,129 @@
+//! 2. Active the feature `auto-detect-format`.
+//!
+//! ```rs
+//! use cargo_packager_resource_resolver::{resources_dir, current_format};
+//!
+//! let resource_path = resources_dir(current_format()).unwrap();
+//! ```
+//!
+use error::Result;
+use std::{env, path::PathBuf};
+
+mod error;
+
+pub use error::Error;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum PackageFormat {
+    /// When no format is used (`cargo run`)
+    None,
+    /// The macOS application bundle (.app).
+    App,
+    /// The macOS DMG package (.dmg).
+    Dmg,
+    /// The Microsoft Software Installer (.msi) through WiX Toolset.
+    Wix,
+    /// The NSIS installer (.exe).
+    Nsis,
+    /// The Linux Debian package (.deb).
+    Deb,
+    /// The Linux AppImage package (.AppImage).
+    AppImage,
+}
+
+/// Get the current package format.
+/// Can only be used if the app was build with cargo-packager
+/// and when the `before-each-package-command` Cargo feature is enabled.
+#[cfg(feature = "auto-detect-format")]
+#[must_use]
+pub fn current_format() -> PackageFormat {
+    // sync with PackageFormat::short_name function of packager crate
+    // maybe having a special crate for the Config struct,
+    // that both packager and resource-resolver could be a
+    // better alternative
+    if cfg!(CARGO_PACKAGER_FORMAT = "app") {
+        PackageFormat::App
+    } else if cfg!(CARGO_PACKAGER_FORMAT = "dmg") {
+        PackageFormat::Dmg
+    } else if cfg!(CARGO_PACKAGER_FORMAT = "wix") {
+        PackageFormat::Wix
+    } else if cfg!(CARGO_PACKAGER_FORMAT = "nsis") {
+        PackageFormat::Nsis
+    } else if cfg!(CARGO_PACKAGER_FORMAT = "deb") {
+        PackageFormat::Deb
+    } else if cfg!(CARGO_PACKAGER_FORMAT = "appimage") {
+        PackageFormat::AppImage
+    } else {
+        PackageFormat::None
+    }
+}
+
+/// Retrieve the resource path of your app, packaged with cargo packager.
+///
+/// ## Example
+///
+/// ```
+/// use cargo_packager_resource_resolver::{resources_dir, PackageFormat};
+///
+/// let resource_path = resources_dir(PackageFormat::Nsis).unwrap();
+/// ```
+///
+pub fn resources_dir(package_format: PackageFormat) -> Result<PathBuf> {
+    match package_format {
+        PackageFormat::None => {
+            env::current_dir().map_err(|e| Error::Io("Can't access current dir".to_string(), e))
+        }
+        PackageFormat::App | PackageFormat::Dmg => {
+            let exe = current_exe()?;
+            let exe_dir = exe.parent().unwrap();
+            exe_dir
+                .join("../Resources")
+                .canonicalize()
+                .map_err(|e| Error::Io("".to_string(), e))
+        }
+        PackageFormat::Wix | PackageFormat::Nsis => {
+            let exe = current_exe()?;
+            let exe_dir = exe.parent().unwrap();
+            Ok(exe_dir.to_path_buf())
+        }
+        PackageFormat::Deb | PackageFormat::AppImage => {
```

Comment:
we should port the implementation for this from https://github.com/tauri-apps/tauri/blob/97e334129956159bbd60e1c531b6acd3bc6139a6/core/tauri-utils/src/platform.rs#L226

Reply:
> When running the app from
/// `src-tauri/target/(debug|release)/`, the path is
/// `${exe_dir}/../lib/${exe_name}`.

I think this case is covered by Package::None so didn't added it.

Also, seems like the package_info.package_name variable is used in camel case on Linux: https://github.com/tauri-apps/tauri/blob/97e334129956159bbd60e1c531b6acd3bc6139a6/core/tauri-utils/src/lib.rs#L58.

So, maybe 
```
let binary_name = exe.file_name().unwrap().to_string_lossy();
```
will not work if it's not camel case, but we could maybe use file_name function and then convert in camel case without the user noticing.

Reply:
i've done it